python_modules/dagster/dagster/_core/definitions/decorators/asset_decorator.py

```diff
@@ -1220,21 +1266,33 @@ def _type_check_deps_and_non_argument_deps(
         check.set_param(non_argument_deps, "non_argument_deps", of_type=(AssetKey, str))
         upstream_asset_deps = list(non_argument_deps)
 
-    return upstream_asset_deps
+    return _make_asset_deps(upstream_asset_deps)
 
 
-def _make_asset_keys(
-    deps: Optional[Iterable[Union[CoercibleToAssetKey, AssetsDefinition, SourceAsset]]]
-) -> Optional[Set[AssetKey]]:
-    """Convert all items to AssetKey in a set. By putting all of the AssetKeys in a set, it will also deduplicate them."""
-    if deps is None:
-        return deps
-
-    deps_asset_keys: Set[AssetKey] = set()
-    for dep in deps:
-        deps_asset_keys.add(AssetKey.from_coercible_or_definition(dep))
+def _make_asset_deps(
+    deps: Optional[Iterable[Union[AssetDep, CoercibleToAssetKey, AssetsDefinition, SourceAsset]]]
+) -> Optional[Iterable[AssetDep]]:
+    with disable_dagster_warnings():
+        if deps is None:
+            return None
+        dep_dict = {}
+        for dep in deps:
+            if not isinstance(dep, AssetDep):
```

Comment:
could do one-liner `asset_dep = dep if isinstance(dep, AssetDep) else AssetDep(dep)`

Reply:
will do to resolve here, but it will get further consolidated in the stacked PR https://github.com/dagster-io/dagster/pull/16432/files#diff-0fcad2d82f20f3987322def0a04db6e04bc81b9777e5ced50f0fed5e527872bdR1268